src/borg/archiver/extract_cmd.py

```diff
@@ -56,33 +55,50 @@ def do_extract(self, args, repository, manifest, archive):
         else:
             pi = None
 
-        for item in archive.iter_items(filter):
-            archive.preload_item_chunks(item, optimize_hardlinks=True)
+        for item in archive.iter_items(None):
             orig_path = item.path
-            if strip_components:
-                item.path = os.sep.join(orig_path.split(os.sep)[strip_components:])
-            if not args.dry_run:
-                while dirs and not item.path.startswith(dirs[-1].path):
-                    dir_item = dirs.pop(-1)
-                    try:
-                        archive.extract_item(dir_item, stdout=stdout)
-                    except BackupError as e:
-                        self.print_warning_instance(BackupWarning(remove_surrogates(dir_item.path), e))
-            if output_list:
```

Comment:
Why did you remove checking output_list?

Reply:
I thought it simplified the logging logic. In the updated code, we always log the paths of the items with a prefix indicating whether they are matched or not. If it is necessary to conditionally control the logging based on the output_list flag, I can reintroduce that check.

Reply:
@ThomasWaldmann Yes it is. However, I have still added the output_list as per your request. Please let me know what else is required to finally close this?